includes/class-sensei-quiz.php

```diff
@@ -1451,6 +1451,18 @@ public static function the_user_status_message( $quiz_id ) {
 
 		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Output escaped above.
 		echo $message;
+
+		// show incomplete questions info message if there are incomplete questions and is preview.
+		if ( is_preview() ) {
+			$lesson_id          = Sensei()->quiz->get_lesson_id( $quiz_id );
+			$all_questions      = Sensei()->quiz->get_questions( Sensei()->lesson->lesson_quizzes( $lesson_id ) );
+			$filtered_questions = Sensei()->quiz->get_questions( Sensei()->lesson->lesson_quizzes( $lesson_id ), 'any', 'meta_value_num title', 'ASC', true );
+
+			if ( count( $all_questions ) !== count( $filtered_questions ) ) {
+				Sensei()->notices->add_notice( __( 'One or more questions in this lesson are incomplete. Incomplete questions are only displayed in preview mode and will not be displayed to students.', 'sensei-lms' ), 'info', 'incomplete-keys-messages' );
+				Sensei()->notices->maybe_print_notices();
+			}
+		}
```

Comment:
I'm not sure if this is the right place for this code. You could get inspiration from the other notices, [for example](https://github.com/Automattic/sensei/search?q=maybe_add_quiz_results_notice).